toast-tk-web-agent/src/main/java/io/toast/tk/agent/web/ScriptInjector.java

```diff
@@ -29,27 +30,28 @@
 	private JFrame frmOpt; 
 	private IAgentServer server;
 	private UriChangeListener uriChangeListener;
-	private WebConfigProvider configProvider;
-	
+	private AgentConfigProvider configProvider;
+	private volatile boolean recording;
+
 	@Inject
-	public ScriptInjector(IAgentServer server, UriChangeListener uriChangeListener, WebConfigProvider configProvider){
+	public ScriptInjector(IAgentServer server, UriChangeListener uriChangeListener, AgentConfigProvider configProvider){
 		this.server = server;
 		this.uriChangeListener = uriChangeListener;
 		this.configProvider = configProvider;
+		this.recording = true;
 	}
 	
 	private Thread initInjectionRetryThread() {
 		return new Thread(new Runnable() {
 			@Override
 			public void run() {
-				while (true) {
+				while (recording) {
 					try {
-						
 						Thread.sleep(1000);
 					} catch (InterruptedException e1) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Either re-interrupt this method or rethrow the "InterruptedException". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS2142)